arviz/data/io_pyro.py

```diff
@@ -89,18 +89,23 @@ def sample_stats_to_xarray(self):
         # extract log_likelihood
         dims = None
         if self.observations is not None and len(self.observations) == 1:
-            obs_name = list(self.observations.keys())[0]
-            samples = self.posterior.get_samples(group_by_chain=False)
-            predictive = self.pyro.infer.Predictive(self.model, samples)
-            obs_site = predictive.get_vectorized_trace(*self._args, **self._kwargs).nodes[obs_name]
-            log_likelihood = obs_site["fn"].log_prob(obs_site["value"]).detach().cpu().numpy()
-            if self.dims is not None:
-                coord_name = self.dims.get("log_likelihood", self.dims.get(obs_name))
-            else:
-                coord_name = None
-            shape = (self.nchains, self.ndraws) + log_likelihood.shape[1:]
-            data["log_likelihood"] = np.reshape(log_likelihood, shape)
-            dims = {"log_likelihood": coord_name}
+            try:
+                obs_name = list(self.observations.keys())[0]
+                samples = self.posterior.get_samples(group_by_chain=False)
+                predictive = self.pyro.infer.Predictive(self.model, samples)
+                vectorized_trace = predictive.get_vectorized_trace(*self._args, **self._kwargs)
+                obs_site = vectorized_trace.nodes[obs_name]
+                log_likelihood = obs_site["fn"].log_prob(obs_site["value"]).detach().cpu().numpy()
+                if self.dims is not None:
+                    coord_name = self.dims.get("log_likelihood", self.dims.get(obs_name))
+                else:
+                    coord_name = None
+                shape = (self.nchains, self.ndraws) + log_likelihood.shape[1:]
+                data["log_likelihood"] = np.reshape(log_likelihood, shape)
+                dims = {"log_likelihood": coord_name}
+            except:
```

Comment:
Can the exception type vary? If possible, it would be great to explicit it, otherwise a pylint exception is neede

Reply:
Thanks! I thought that the tests have passed... my bad

Reply:
@OriolAbril The broadcasting issue can happen at multiple paths of Pyro code base so I choose the easier way: to add pylint exception here. Thanks for your suggestion!